app/facades/foods_facade.rb

```diff
@@ -0,0 +1,13 @@
+class FoodsFacade
+  def self.foods(ingredient)
+    foods = FoodsService.foods_search(ingredient)
+    foods[:foods].map do |food|
+      Food.new(food)
+    end[0..9]
+  end
+
+  def self.results_number(ingredient)
+    foods = FoodsService.foods_search(ingredient)
+    foods[:totalHits]
+  end
```

Comment:
As I said, I'm glad you broke these requirements up. However, we don't want to hit our external api service more than once considering it only takes one request/response cycle to get what we need. So, you have some options. You could:

1. Combine this functionality into one method, and return a hash with both the totalHits and food objects within that hash. Essentially only ever calling `::food_search` once. 
2. Cache the value of the service method `::food_search`
3. Change the `::food_search` class method to an instance method `#food_search` and then memoize that value.